bayesian/utility/security_vendor.py

```diff
@@ -0,0 +1,355 @@
+# Copyright © 2020 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# Author: Deepak Sharma <deepshar@redhat.com>
+#
+"""Utility function from API v2."""
+
+from urllib.parse import quote
+import logging
+from bayesian.utility.db_gateway import GraphAnalyses
+import semantic_version as sv
+
+
+logger = logging.getLogger(__file__)
+
+
+def version_info_tuple(version):
+    """Return version information in form of (major, minor, patch, build) for a given Version.
+
+    : type version: semantic_version.base.Version
+    : param version: The semantic version whole details are needed.
+    : return: A tuple in form of Version.(major, minor, patch, build)
+    """
+    if type(version) == sv.base.Version:
+        return (version.major,
+                version.minor,
+                version.patch,
+                version.build)
+    return (0, 0, 0, tuple())
+
+
+def convert_version_to_proper_semantic(version, package_name=None):
+    """Perform Semantic versioning.
+
+    : type version: string
+    : param version: The raw input version that needs to be converted.
+    : type return: semantic_version.base.Version
+    : return: The semantic version of raw input version.
+    """
+    conv_version = sv.Version.coerce('0.0.0')
+    try:
+        if version in ('', '-1', None):
+            version = '0.0.0'
+        """Needed for maven version like 1.5.2.RELEASE to be converted to
+        1.5.2 - RELEASE for semantic version to work."""
+        version = version.replace('.', '-', 3)
+        version = version.replace('-', '.', 2)
+        # Needed to add this so that -RELEASE is account as a Version.build
+        version = version.replace('-', '+', 3)
+        conv_version = sv.Version.coerce(version)
+    except ValueError:
+        logger.error(
+            f"Unexpected ValueError for the package {package_name} due to version {version}")
+        pass
+    finally:
+        return conv_version
+
+
+class VendorAnalyses:
```

Comment:
rename this to componentAnalysis

Reply:
@prashbnair we already have a same-named class...
https://github.com/fabric8-analytics/fabric8-analytics-server/blob/43fe152b7c7217b87955b3a5c854d221957deace/bayesian/api/api_v2.py#L94

Reply:
can the one in api be renamed to ComponentAnalysesResource

Reply:
Going ahead with `ComponentAnalysesApi `